### ISSUE TITLE
LSP server running but can't get it to work

### ISSUE THREAD

**Issue by htetyan** [merrickluo/lsp-tailwindcss#47]
**make sure to read the [troubleshooting](https://github.com/merrickluo/lsp-tailwindcss#troubleshooting) guide first.**

Been trying to figure out how to get this going without any success. 

I've read through the other activation issues and all the previous closed issues on this. Haven't seem to got it running here. Using Doom Emacs.

The lsp server is installed:
`lsp-describe-session` shows that the tailwindcss server is running on the project folder but I'm not getting any auto-completion inside my html file in web-mode. 

I've tested this file and folder with VS Code and there is no problem with the tailwind setup it on that editor. 

Also added the `use-package` config to my `config.el` that I saw in other closed issues.

``` elisp 
(use-package! lsp-tailwindcss
  :init
  (setq lsp-tailwindcss-add-on-mode t)
  :config
  (setq lsp-tailwindcss-major-modes '(svelte-mode html-mode sgml-mode mhtml-mode web-mode css-mode)))
```
Edit: 
I also get a warning when it loads 
`Warning (lsp-mode): Unknown notification: @/tailwindCSS/error`

Edit 2:
Replicated the same error and issue on another device running the same config.

Here is a screenshot:

<img width="1116" alt="Screen Shot 2022-10-22 at 21 57 24" src="https://user-images.githubusercontent.com/3907491/197343017-69dc7dd4-2075-48c6-b84b-0f98b4fec973.png">

**Comment by year39** [comment]
Just ran into the same thing, just make sure your workspace is configured to use [tailwind](https://tailwindcss.com/docs/installation#create-your-configuration-file).

**Comment by htetyan** [comment]
> Just ran into the same thing, just make sure your workspace is configured to use [tailwind](https://tailwindcss.com/docs/installation#create-your-configuration-file).

Thank you for replying, been trying to solve and hitting a wall.

@year39 Did you get the error buffer warning? 
`Warning (lsp-mode): Unknown notification: @/tailwindCSS/error`

<img width="909" alt="Screen Shot 2022-10-26 at 23 13 26" src="https://user-images.githubusercontent.com/3907491/198065354-adf2cb79-66b4-4049-93c1-a6c2aa3f0dfd.png">

This is my configuration:
<img width="650" alt="Screen Shot 2022-10-26 at 23 04 11" src="https://user-images.githubusercontent.com/3907491/198063394-99d8ecd2-177b-492a-b686-97f85c27f72a.png">

The config is working in VS Code:
<img width="940" alt="Screen Shot 2022-10-26 at 23 07 46" src="https://user-images.githubusercontent.com/3907491/198064014-affa8805-b55d-4d22-adda-fa43ad5a743a.png">

Please let me know if there something wrong with this tailwindcss config.

**Comment by htetyan** [comment]
**_Update and closing issue_**

**TLDR**
Was fixed by rolling back to tailwindcss 3.1.8 npm package. 

**Original issue:** 
Server is running but does not activate auto-compeletion. A popup buffer shows the following message: `Warning (lsp-mode): Unknown notification: @/tailwindCSS/error`

**De-bugging with LSP Log**

Checked the *lsp-log* buffer and output:

```
Command "node /Users/../.emacs.d/.local/etc/lsp/tailwindcss/extension/dist/tailwindServer.js --stdio" is present on the path.
...
...
Creating watchers for following 3 folders:
  /Users/../Projects/tailwind-2
  /Users/../Projects/tailwind-2/.log
  /Users/../Projects/tailwind-2/src
Found Tailwind CSS config file: /Users/../Projects/tailwind-2/tailwind.config.js
Loaded postcss v8.4.18: /Users/../Projects/tailwind-2/node_modules/postcss
Loaded tailwindcss v3.2.1: /Users/../Projects/tailwind-2/node_modules/tailwindcss

**Tailwind CSS: Cannot read properties of undefined (reading 'modifier')
TypeError: Cannot read properties of undefined (reading 'modifier')
    at api.addVariant (/Users/../Projects/tailwind-2/node_modules/tailwindcss/lib/lib/setupContextUtils.js:578:40)
    at /Users/../Projects/tailwind-2/node_modules/tailwindcss/lib/lib/setupContextUtils.js:538:38
    at /Users/../.emacs.d/.local/etc/lsp/tailwindcss/extension/dist/tailwindServer.js:695:4431
    at Array.forEach (<anonymous>)
    at Jwe (/Users/../.emacs.d/.local/etc/lsp/tailwindcss/extension/dist/tailwindServer.js:695:3916)
    at /Users/../.emacs.d/.local/etc/lsp/tailwindcss/extension/dist/tailwindServer.js:695:286
    at Generator.next (<anonymous>)
    at i (/Users/../.emacs.d/.local/etc/lsp/tailwindcss/extension/dist/tailwindServer.js:1:1059)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)**
```

Searched for issues relating to `Tailwind CSS: Cannot read properties of undefined (reading 'modifier')`

Found the following issues on github:
https://github.com/tailwindlabs/tailwindcss-intellisense/issues/641
https://github.com/tailwindlabs/tailwindcss/discussions/9641

**Fix:**
Rolling back the npm package of tailwindcss to [3.1.8](https://www.npmjs.com/package/tailwindcss/v/3.1.8)


<img width="648" alt="Screen Shot 2022-10-26 at 23 39 46" src="https://user-images.githubusercontent.com/3907491/198073696-b596977d-02d6-4e33-af66-6d18ea8f6883.png">
